serverless/event-gateway
# Add AWS Firehose provider

## What did you implement:

Added [AWS Firehose](https://aws.amazon.com/kinesis/data-firehose/) provider.

## How did you implement it:

Very similar to the Kinesis implementation from #392 

## How can we verify it:

1. Create a Firehose stream in AWS.

2. Create a Firehose "function" by calling the `registerFunction` API. Payload body should be:

```
{
    "functionId": "firehose-stream",
    "type": "awsfirehose",
    "provider": {
        "region": "<region-of-stream>",
        "awsAccessKeyId": "<access-key-with PutRecord permission>",
        "awsSecretAccessKey": "<secret-access-key>",
        "deliveryStreamName": "<name-of-stream>"
    }
}
```

3. Create a subscription with the `createSubscription` API with the following payload:

```
{
	"functionId": "firehose-stream", 
	"event": "testFirehoseEvent"
}
```

4. Invoke your event by calling the Events API with a header of `event: testFireshoseEvent` and a payload body that you wanted inserted to Firehose.

5. Check your Firehose stream destination to make sure your record made it there. 

## Todos:

- [X] Provide verification commands / resources

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
